### ISSUE TITLE
Project dependencies not working when using an authentication setup

### ISSUE THREAD

**Issue by jzaratei** [vitalets/playwright-bdd#32]
I'm trying to implement authentication setup using a project dependency as it is shown in this doc https://playwright.dev/docs/auth

So for that I've created an auth.setup.ts file that includes the loging process and then I configured the playwright.config.ts with:

```
  projects: [
    { 
      name: 'setup',
      testMatch: '**/*setup.ts' 
    },
    {
      name: 'Regresion tests using chromium',
      dependencies: ['setup'],
      use: { 
        ...devices['Desktop Chrome'],
        storageState: STORAGE_STATE, 
      },
    },
```
When I run my test cases I get this error:
Error: Error reading storage state from myprojectdir\\.auth\user.json: ENOENT: no such file or directory

It seems that the project-dependency is not triggered.

**Comment by NikkTod** [comment]
In my setup it is working, seems you have some problems with the path of the json.

Do you have these lines in the beginning of the config file, in your case for STORAGE_STATE

![image](https://github.com/vitalets/playwright-bdd/assets/94455079/ea035a9a-57b9-46a6-8259-dd80a9f1287c)

After Login is successfull and you wait for some element to be visible you store the session right?

![image](https://github.com/vitalets/playwright-bdd/assets/94455079/d5ca0a24-39c6-4394-b651-373cdcc87734)

Do you see the json file being created after you login the first time?

**Comment by jzaratei** [comment]
@NikkTod yeah, I have` export const STORAGE_STATE = path.join(__dirname, '.auth/user.json');`
but when I run the tests I'm just getting this error 
`Error: Error reading storage state from myprojectdir\.auth\user.json: ENOENT: no such file or directory`
nothing else. Actually, the user.json, where is suppose to be stored the state, never gets to be created.
it seems that my auth.setup.ts is never triggered:
```

import { test as setup } from '@playwright/test';
import { STORAGE_STATE } from './playwright.config';

setup('authenticate', async ({ page }) => {
    await page.goto('https://myapp/');
    await page.locator('a', { hasText: 'External Identities'} ).click();
    await page.getByPlaceholder('Email Address').fill('****');
    await page.getByPlaceholder('Password').fill('***');
    await page.getByRole('button', { name: 'Sign in' }).click();
    await page.context().storageState({ path: STORAGE_STATE });
});
```

**Comment by Ridd0** [comment]
Dependencies don't work if `require` or `import` is identical in both projects. I have a repo where steps are more general, used by a couple of features and unfortunately, it is just not working.

I made a quick PoC, and copy all my general steps to another folder, so I have `steps/` and `steps-copy/`, and by changing `import` (or `require`) in the second project it started to work, but I feel its not the best practise to have duplicated steps.

**Comment by vitalets** [comment]
Thank you @Ridd0 I will check.
Reopening this issue.

**Comment by vitalets** [comment]
Fixed in v4.0.0.
@Ridd0 @jzaratei could you re-check on `playwright-bdd@4.0.0` and share the results?

**Comment by jzaratei** [comment]
I though this issue was close, so, I created a new one https://github.com/vitalets/playwright-bdd/issues/39 with the results
